s3torchconnector/src/s3torchconnector/dcp/fsdp_filesystem.py

```diff
@@ -0,0 +1,220 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+import io
+import logging
+import os
+from contextlib import contextmanager
+from typing import Generator, Union, Optional
+
+from torch.distributed.checkpoint.filesystem import (
+    FileSystem,
+    FileSystemReader,
+    FileSystemWriter,
+)
+
+from s3torchconnector import S3Checkpoint
+from s3torchconnector._s3client import S3Client
+from s3torchconnector._s3dataset_common import parse_s3_uri  # type: ignore
+from s3torchconnectorclient._mountpoint_s3_client import S3Exception
+
+logger = logging.getLogger(__name__)
+
+
+class S3FileSystem(FileSystem):
```

Comment:
1. Renamed this `S3FS` ➡️ `S3FileSystem`, since s3fs is already an existing lib ([fsspec/s3fs](https://github.com/fsspec/s3fs)).